crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -307,26 +307,24 @@ def deployment_check
         ret = {}
         # Make sure that node with nova-compute is not upgraded before nova-controller
         nova_order = BarclampCatalog.run_order("nova")
-        ["kvm", "xen"].each do |virt|
-          NodeObject.find("roles:nova-compute-#{virt}").each do |node|
-            # nova-compute with nova-controller on one node is not non-disruptive,
-            # but at least it does not break the order
-            next if node.roles.include? "nova-controller"
-            next if ret.any?
-            wrong_roles = []
-            node.roles.each do |role|
-              # these storage roles are handled separately
-              next if ["cinder-volume", "swift-storage"].include? role
-              # compute node roles are fine
-              next if role.start_with?("nova-compute") || role == "pacemaker-remote"
-              r = RoleObject.find_role_by_name(role)
-              next if r.proposal?
-              b = r.barclamp
-              next if BarclampCatalog.category(b) != "OpenStack"
-              wrong_roles.push role if BarclampCatalog.run_order(b) < nova_order
-            end
-            ret = { controller_roles: { node: node.name, roles: wrong_roles } } if wrong_roles.any?
+        NodeObject.find("roles:nova-compute-*").each do |node|
```

Comment:
I think that this check is for making sure that at least **disruptive** upgrade can work - during which we have an algorithm to pick the nodes order. During disruptive (normal) upgrade mode, we have to upgrade non-kvm nodes as well, but we still need that right order that would be broken if `nova-compute` would go before `nova-controller`.

So this change is actually fixing bug that was in the code already before.